src/main/java/pl/agh/edu/master_diet/service/BrowserService.java

```diff
@@ -0,0 +1,58 @@
+package pl.agh.edu.master_diet.service;
+
+import pl.agh.edu.master_diet.core.model.database.Browsable;
+import pl.agh.edu.master_diet.repository.BrowserRepository;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+public class BrowserService<T extends Browsable> {
+
+    private static final int ALLOW_SINGLE_TYPO_MINIMUM_WORD_LENGTH = 5;
+
+    private final BrowserRepository<T> browserRepository;
+
+    public BrowserService(BrowserRepository<T> browserRepository) {
```

Comment:
Lombok might be used instead of this explicit constructor

Reply:
It's not a spring component. There is no way to autowired generic components. It created by classic constructor.

Reply:
???

Reply:
```@AllArgsConstructor``` or ```@RequiredArgsConstructor```?

Reply:
there is no @Componet or anything like this. I have no idea how to autowired generic repository into a generic class. Stackoverflow said that there is no way to do this so I did initialization by the constructor.

Reply:
Changed